app/views/devise/registrations/new.html.erb

```diff
@@ -14,10 +14,9 @@
   </div>
 
   <div class="field">
-    <%= f.fields_for :department do |d| %>
+    <%= f.fields_for :department_id do |d| %>
```

Comment:
Yeah, it's definitely because of the hash being submitted, and the hash is happening because you're doing the `fields_for` function.  If you don't do that, and just do `f.label :department_id`, it should work.

Let me know how that works for you.